### PR TITLE
Test: (Tree, Treepicker, Uploader, Whisper): Removing the getDOMNode and using render

### PR DESCRIPTION
**Pull request by Chriscaracach**
Why I am doing this: https://github.com/rsuite/rsuite/discussions/3200
What I did: removing the getDOMNode and using the render in @testing-library/react

This update involves 4 component changes: Tree, Treepicker, Uploader, Whisper

### CODE REVIEW

**Comment by Chriscaracach** [comment]
First Open Source contribution. Any feedback is welcome. Thanks for the oportunity

File: src/TreePicker/test/TreeNodeSpec.tsx [hunk 1]
```diff
@@ -1,60 +1,66 @@
...
     const onDragStartSpy = sinon.spy();
-    const instance = getDOMNode(
+    const { container } = render(
       <TreeNode layer={0} onDragStart={onDragStartSpy} nodeData={1} visible />
     );
-
-    Simulate.dragStart(instance);
+    //eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+    const div = container.querySelector('div') as HTMLElement;
+    Simulate.dragStart(div); //Not working
```
Comment (SevenOutman) [inline src/TreePicker/test/TreeNodeSpec.tsx:22]: What does "Not working" mean here?

File: src/TreePicker/test/TreeNodeSpec.tsx [hunk 2]
```diff
@@ -1,60 +1,66 @@
...
   it('Should call `onDragStart` callback', () => {
     const onDragStartSpy = sinon.spy();
-    const instance = getDOMNode(
+    const { container } = render(
       <TreeNode layer={0} onDragStart={onDragStartSpy} nodeData={1} visible />
     );
-
-    Simulate.dragStart(instance);
+    //eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+    const div = container.querySelector('div') as HTMLElement;
```
Comment (SevenOutman) [inline src/TreePicker/test/TreeNodeSpec.tsx:21]: This query returns the root element all the same. Use `container.firstChild` (also remove `eslint-disable` annotation then).

File: src/Uploader/test/UploadFileItemSpec.tsx [hunk 3]
```diff
@@ -14,32 +14,31 @@ const file = {
...
   it('Should output a UploadFileItem', () => {
-    const instance = getDOMNode(<UploadFileItem file={file} />);
-    assert.include(instance.className, 'rs-uploader-file-item');
+    const { container } = render(<UploadFileItem file={file} />);
+    expect(container.firstChild).to.have.class('rs-uploader-file-item');
   });
 
   it('Should render picture-text for layout', () => {
-    const instance = getDOMNode(<UploadFileItem listType="picture-text" file={file} />);
-    assert.include(instance.className, 'rs-uploader-file-item-picture-text');
```
Comment (SevenOutman) [inline src/Uploader/test/UploadFileItemSpec.tsx:24]: This assertion is missing from the updated test.

File: src/Uploader/test/UploadTriggerSpec.tsx [hunk 4]
```diff
@@ -1,87 +1,98 @@
...
+    ReactTestUtils.Simulate.change(container.querySelector('input') as HTMLElement);
 
     expect(onChange).to.have.been.calledOnce;
   });
 
   it('Should have a name', () => {
-    const instance = getDOMNode(<UploadTrigger name="file" />);
-    assert.ok(instance.querySelector('input[name="file"]'));
+    const { container } = render(<UploadTrigger name="file" />);
+    console.log(container);
```
Comment (SevenOutman) [inline src/Uploader/test/UploadTriggerSpec.tsx:48]: Remove `console.log`.

File: src/Uploader/test/UploadTriggerSpec.tsx [hunk 5]
```diff
@@ -1,87 +1,98 @@
...
+    // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+    assert.equal((container.querySelector('button') as HTMLElement).style.fontSize, fontSize);
   });
 
   it('Should have a custom className prefix', () => {
-    const instance = getDOMNode(<UploadTrigger classPrefix="custom-prefix" />);
-    assert.ok(instance.className.match(/\bcustom-prefix\b/));
+    const { container } = render(<UploadTrigger classPrefix="custom-prefix" />);
+    // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+    assert.ok(container.querySelector('.rs-custom-prefix')?.className.match(/\bcustom-prefix\b/));
```
Comment (SevenOutman) [inline src/Uploader/test/UploadTriggerSpec.tsx:68]: Use `contianer.firstChild`. You may also want to use `expect(...).to.have.class(regex)` assertion :)

**Review by SevenOutman — CHANGES_REQUESTED** [review]
Hi @Chriscaracach, thanks for contributing this edit! Good completeness overall! And here's my review feedback that I would like you to look into.

File: src/Uploader/test/UploadFileItemSpec.tsx (hunk 3 above)
Comment (Chriscaracach) [reply src/Uploader/test/UploadFileItemSpec.tsx:24]: That's strange. I don't see that on my local. Pushing it again.

File: src/TreePicker/test/TreeNodeSpec.tsx [hunk 6]
```diff
@@ -1,60 +1,66 @@
...
   });
 
   it('Should call `onDragEnter` callback', () => {
     const onDragEnterSpy = sinon.spy();
-    const instance = getDOMNode(
+    const { container } = render(
       <TreeNode layer={0} onDragEnter={onDragEnterSpy} nodeData={1} visible />
     );
+    //eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+    const div = container.querySelector('div') as HTMLElement;
```
Comment (SevenOutman) [inline src/TreePicker/test/TreeNodeSpec.tsx:34]: Replace this query with `container.firstChild`. Also update other similar queries in other test cases in this test suite.

File: src/Uploader/test/UploadFileItemSpec.tsx (hunk 3 above)
Comment (SevenOutman) [inline src/Uploader/test/UploadFileItemSpec.tsx:24]: This assertion is still missing. Same for the first assertion in the next test case.

**Review by SevenOutman — CHANGES_REQUESTED** [review]

**Review by SevenOutman — APPROVED** [review]
LGTM now!

**Pull request merged**